### PR TITLE
fix(API): Revert breaking change vs. 1.4 regarding forcing the Content-Length header

### PR DESCRIPTION
**Pull request by kgriffs**

### CODE REVIEW

**Comment by kgriffs** [comment]
This patch reverts the 2.0 behavior to that of 1.4. I found this issue while testing a project that was relying on the Content-Length being overridden. It resulted in a lot of weird behavior by HAProxy / NGINX that is difficult to diagnose.

File: tests/test_headers.py [hunk 1]
```diff
@@ -207,14 +207,24 @@ def on_get(self, req, resp):
...
-        resp.body = self._body
+
+        if self._body:
+            resp.body = self._body
+
+        if self._data:
+            resp.data = self._data
+
+    def on_head(self, req, resp):
+        # NOTE(kgriffs): Use stream_len for now to cover the deprecated alias
```
Comment (vytas7) [inline tests/test_headers.py]: Intent of this note is somewhat unclear (suspected copy-paste from L216?)

**Comment by vytas7** [comment]
What is even worse, reverse-proxies, front-end servers or HTTP clients might hang for a long time, or time-out waiting for "full" content if you set the `Content-Length` to a greater value than the actual content length being sent.

File: tests/test_headers.py (hunk 1 above)
Comment (kgriffs) [reply tests/test_headers.py]: oops, yeah. Copy-paste #fail - Let me fix that real quick

Comment (kgriffs) [reply tests/test_headers.py]: fixed

**Pull request merged**